internal/vault/testutils_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/hashicorp/vault/api"
+)
+
+// TestHTTPServer creates a test HTTP server that handles requests until
+// the listener returned is closed.
+// XXX: based off of github.com/hashicorp/vault/api/client_test.go
+func NewTestHTTPServer(t *testing.T, handler http.Handler) (*api.Config, net.Listener) {
+	t.Helper()
+
+	server, ln, err := testHTTPServer(handler, nil)
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	go server.Serve(ln)
+
+	config := api.DefaultConfig()
+	config.Address = fmt.Sprintf("http://%s", ln.Addr())
+
+	return config, ln
+}
+
+func testHTTPServer(handler http.Handler, tlsConfig *tls.Config) (*http.Server, net.Listener, error) {
+	ln, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		return nil, nil, err
+	}
+
+	server := &http.Server{
+		Handler:   handler,
+		TLSConfig: tlsConfig,
+	}
+
+	return server, ln, err
+}
```

Comment:
I'm curious why this function is split out from `NewTestHTTPServer()`, since I don't see it called independently.

Reply:
I can probably move it back to NewTestHTTPServer(), I left it here in case we want to setup an HTTPTestServer.

Reply:
Decided to leave it as-is.